### ISSUE TITLE
Installer doesn't boot on Hyper-V

### ISSUE THREAD

**Issue by puneetse** [clearlinux/distribution#1545]
**Describe the bug**
The live-server.iso does not boot on Hyper-V and gets stuck at "Searching for installer media, retrying...".

**To Reproduce**
1. Create Hyper-V VM
2. Add a DVD Drive to the VM
3. Mount live-server.iso (tested with 30210 and 29690)
4. Try to boot from it.

**Expected behavior**
The installer ISO should boot under Hyper-V VMs

**Additional context**
The bootloader appears so the DVD drive is there. On a working Hyper-V VM with CL, the same device shows up as `/dev/sr0` with the `CLR_ISO` label when the ISO is mounted.

**Comment by mdhorn** [comment]
If we want to support Hyper-V it appears that support needs to be added to kernel-native. `linux-hyperv`
@ahkok @bryteise Is this something we should be supporting?

@miguelinux Is this reasonable to enable in our native kernel? And the LTS kernels? Or just native to allow installs and then a special kernel for hyper-v?

**Comment by ahkok** [comment]
We have a `-hyperv` specific kernel. This is likely required?

**Comment by mdhorn** [comment]
There is already a published [Hyper-V image](https://cdn.download.clearlinux.org/releases/31770/clear/clear-31770-hyperv.vhdx.gz) but I'm assuming this is meeting the needs? @puneetse

**Comment by puneetse** [comment]
Right, the hyperv image works. This request is to enable the ISO installer for installing a Clear Linux VMfrom scratch, like is possible for vmware/virtualbox.

**Comment by mdhorn** [comment]
To enable the default installer would first require that the native-kernel needs to have the linux-hyperv enabled. If this is approved and enabled, we could next look at adding the hyper-v kernel to the installer. If there a method to detect that the VM is hyper-v we could auto-select that kernel similar to how we select the kernel-lts for VirtualBox.

@ahkok Is there a reason we have a one-off kernel for hyper-v instead of enabling it in the native-kernel and lts-kernels?  If it were supported in native and LTS kernels, this would just work.  Given there is a unique kernel currently for hyper-v, want to make sure we are following a reasonable enablement path.

**Comment by tristanbarcelon** [comment]
I think the same issue be true for the live-desktop iso as well. I'd very much like to install try this distro out in a hyper-v guest vm.

**Comment by mdhorn** [comment]
> I think the same issue be true for the live-desktop iso as well. I'd very much like to install try this distro out in a hyper-v guest vm.

Correct, the problem is in the kernel which is the same for all of the installer images.

You can try Clear Linux OS in Hyper-v right now using the prebuilt Hyper-V image:
https://clearlinux.org/downloads#hyperv

**Comment by dorileo** [comment]
If you're enabling hyper-v on the installer image make sure to also add hyper-v kernel to the list of available kernels.

**Comment by mdhorn** [comment]
> @ahkok Is there a reason we have a one-off kernel for hyper-v instead of enabling it in the native-kernel and lts-kernels? If it were supported in native and LTS kernels, this would just work. Given there is a unique kernel currently for hyper-v, want to make sure we are following a reasonable enablement path.

@ahkok Is there a reason we have a one-off kernel for hyper-v instead of enabling it in the native-kernel and lts-kernels? If it were supported in native and LTS kernels, this would just work. Given there is a unique kernel currently for hyper-v, want to make sure we are following a reasonable enablement path. Thoughts?

**Comment by ahkok** [comment]
All the virtualization kernels are small because they do not need to support lots of legacy hardware.

It's entirely fine to make the -native kernel also support booting in HyperV, permitted that it doesn't impose a significant cost to others (e.g. we don't enable Xen support unless it's the AWS kernel). But other than that, this is likely OK if it doesn't impose.

So, AWS, HyperV, etc. kernels are "expressly optimized for" their purpose.

We should always recommend and install the optimized kernels for the VM where possible (yes, I realize that is likely a feature that isn't present).

**Comment by mdhorn** [comment]
> All the virtualization kernels are small because they do not need to support lots of legacy hardware.
> 
> It's entirely fine to make the -native kernel also support booting in HyperV, permitted that it doesn't impose a significant cost to others (e.g. we don't enable Xen support unless it's the AWS kernel). But other than that, this is likely OK if it doesn't impose.
> 
> So, AWS, HyperV, etc. kernels are "expressly optimized for" their purpose.
> 
> We should always recommend and install the optimized kernels for the VM where possible (yes, I realize that is likely a feature that isn't present).

@miguelinux Could we get your help with enabling this in the native kernel?

**Comment by ahkok** [comment]
https://github.com/clearlinux-pkgs/linux/blob/master/config#L320

It is enabled already. The problem is more likely some HyperV specific virtual device driver.

**Comment by ahkok** [comment]
Scratch that remark:

https://github.com/clearlinux-pkgs/linux/blob/master/config#L6712

**Comment by tristanbarcelon** [comment]
@ahkok I tried using the hyper-v image as instructed since I'm unable to boot the desktop iso under hyper-v. So far it works but when I add the [desktop-autostart bundle](https://clearlinux.org/software/bundle/desktop-autostart) using swupd, I am observing 2 unexpected behaviors:

- an error about /usr not having enough space needed by the bundle
  I reran the bundle addition but told it to ignore the error

- upon restart, I don't see a desktop

Is the hyper-v image not supportive of the desktop bundle due to other drivers being left off of the kernel?

**Comment by puneetse** [comment]
The virtual disk is limited by design and needs to be increased manually through Hyper-V and in the OS. See https://docs.01.org/clearlinux/latest/guides/maintenance/increase-virtual-disk-size.html for how to increase the virtual disk size.

Graphics should work after the bundles are installed but last I checked there were open known issues regarding general Hyper-V + Linux graphics slowness. 

@tristanbarcelon since these are separate problems, if you need more help please create another thread on GitHub or the forums to keep this one on focused.

**Comment by miguelinux** [comment]
The missing modules are already in the -native kernel
clearlinux-pkgs/linux@7fa5785

**Comment by tristanbarcelon** [comment]
Thanks @miguelinux I'm able to install linux-desktop iso build 32600 in HyperV now. Is there anything else I need to do to boot straight to a desktop ui? I specified the desktop bundle during the install process and I can confirm that desktop and desktop-autostart bundles are installed. Each time I start the VM, I'm just booted straight to a console.
